### PR TITLE
Madhura_Assignment_17_45mins

### PR DESCRIPTION
**Pull request by MadhuraMJun20**
indexOf() : 
IndexOfChar

Methods of Character class : 
MethodsOfChar

### CODE REVIEW

File: src/madhuraMJun20/IndexOfChar.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+Hint : To find first index of character --> indexOf() method
+            To find last index of character --> lastIndexOf() method.
+
+*/
+public class IndexOfChar {
+	void firstLastIndex(String str) {
+		for(int index=0; index<str.length();index++) {
+			char ch = str.charAt(index);
+			if(index==str.indexOf(ch)) {
+				if(str.indexOf(ch) !=  str.lastIndexOf(ch)) {
```
Comment (KrishnaKTechnocredits) [inline src/madhuraMJun20/IndexOfChar.java:19]: Perfect logic.

File: src/madhuraMJun20/MethodsOfCharClass.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+		int digitC=0 ; int letterC =0; int upperCase=0; int lowerCase = 0; int spChar=0;
+		
+		for(int index=0;index<name.length();index++) {
+			
+			char ch= name.charAt(index);
+			if(Character.isDigit(ch))
+				digitC++;
+			else if (Character.isLetter(ch)) {
+				letterC++;
+				if(Character.isUpperCase(ch))
```
Comment (KrishnaKTechnocredits) [inline src/madhuraMJun20/MethodsOfCharClass.java]: letterC this variable is not needed dear. 
Think here, you can reduce else if > if and else condition.

Comment (MadhuraMJun20) [reply src/madhuraMJun20/MethodsOfCharClass.java]: Done.

File: src/madhuraMJun20/IndexOfChar.java (hunk 1 above)
Comment (MadhuraMJun20) [reply src/madhuraMJun20/IndexOfChar.java:19]: Thank you

File: src/madhuraMJun20/MethodsOfCharClass.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+				digitC++;
+			else if(Character.isUpperCase(ch))
+					upperCase++;
+			else if(Character.isLowerCase(ch))
+					lowerCase++;
+			else
+				spChar++;			
+		}
+		System.out.println("Total Digits : " +digitC);
+		System.out.println("Total Letters : " +(upperCase+lowerCase));
```
Comment (KrishnaKTechnocredits) [inline src/madhuraMJun20/MethodsOfCharClass.java:30]: Perfect !! Good job.

**Review by KrishnaKTechnocredits — APPROVED** [review]

**Pull request merged**